diff --git a/config/install/loqate.loqateapikeyconfig.yml b/config/install/loqate.loqateapikeyconfig.yml
new file mode 100644
index 0000000..b30e591
--- /dev/null
+++ b/config/install/loqate.loqateapikeyconfig.yml
@@ -0,0 +1 @@
+loqate:
diff --git a/loqate.links.menu.yml b/loqate.links.menu.yml
new file mode 100644
index 0000000..31932cc
--- /dev/null
+++ b/loqate.links.menu.yml
@@ -0,0 +1,7 @@
+loqate.loqate_api_key_config_form:
+  title: 'Loqate Api'
+  route_name: loqate.loqate_api_key_config_form
+  description: 'Loqate api configurations.'
+  parent: system.admin_config_services
+  weight: 99
+
diff --git a/loqate.routing.yml b/loqate.routing.yml
new file mode 100644
index 0000000..7745ccf
--- /dev/null
+++ b/loqate.routing.yml
@@ -0,0 +1,11 @@
+
+loqate.loqate_api_key_config_form:
+  path: '/admin/config/loqate/api'
+  defaults:
+    _form: '\Drupal\loqate\Form\LoqateApiKeyConfigForm'
+    _title: 'LoqateApiKeyConfigForm'
+  requirements:
+    _permission: 'access administration pages'
+  options:
+    _admin_route: TRUE
+
diff --git a/src/Form/LoqateApiKeyConfigForm.php b/src/Form/LoqateApiKeyConfigForm.php
new file mode 100644
index 0000000..714b451
--- /dev/null
+++ b/src/Form/LoqateApiKeyConfigForm.php
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\loqate\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class LoqateApiKeyConfigForm.
+ */
+class LoqateApiKeyConfigForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'loqate.loqateapikeyconfig',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'loqate_api_key_config_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('loqate.loqateapikeyconfig');
+    $form['loqate_api_key'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Loqate Api key'),
+      '#description' => $this->t('Provide the api key from https://account.loqate.com/, Read more from https://www.loqate.com/resources/support/setup-guides/advanced-setup-guide/#creating_a_key'),
+      '#maxlength' => 64,
+      '#size' => 64,
+      '#default_value' => $config->get('loqate_api_key'),
+    ];
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    parent::submitForm($form, $form_state);
+
+    $this->config('loqate.loqateapikeyconfig')
+      ->set('loqate_api_key', $form_state->getValue('loqate_api_key'))
+      ->save();
+  }
+
+}
