diff --git a/loqate.module b/loqate.module
index 50b5987..73cf892 100644
--- a/loqate.module
+++ b/loqate.module
@@ -27,10 +27,12 @@ function loqate_help($route_name, RouteMatchInterface $route_match) {
  * Implements hook_webform_options_alter().
  */
 function loqate_webform_options_alter(array &$options, array &$element) {
-  if (is_array($element['#options'])) {
-    return;
-  }
+  if (isset($element['#options])) {
+    if (is_array($element['#options'])) {
+      return;
+    }
 
-  // Where a system set of options is used, apply a data attribute to use in js.
-  $element['#attributes']['data-option-type'] = $element['#options'];
+    // Where a system set of options is used, apply a data attribute to use in js.
+    $element['#attributes']['data-option-type'] = $element['#options'];
+  }
 }
